src/Init/Prelude.lean

```diff
@@ -2630,14 +2630,20 @@ def Array.empty {α : Type u} : Array α := mkEmpty 0
 def Array.size {α : Type u} (a : @& Array α) : Nat :=
  a.toList.length
 
-/-- Access an element from an array without bounds checks, using a `Fin` index. -/
+/--
+Access an element from an array without needing a runtime bounds checks,
+using a `Nat` index and a proof that it is in bounds.
+
+(Note this function does not have a tactic autoparameter for the proof argument,
+for bootstrapping reasons. Generally, one will use `a[i]` instead.)
```

Comment:
```suggestion
This function does not use `get_elem_tactic` to automatically find the proof that
the index is in bounds. This is because the tactic itself needs to look up values in
arrays. Use the indexing notation `a[i]` instead.
```

I'm not sure the target audience knows the term "bootstrapping" - what do you think?